pkg/minikube/cni/custom.go

```diff
@@ -63,3 +69,12 @@ func (c Custom) Apply(r Runner) error {
 func (c Custom) CIDR() string {
 	return DefaultPodCIDR
 }
+
+func (c Custom) removeCNIConfig(r Runner) error {
+	klog.Infof("Removing existing cni configuration (/etc/cni/net.d/*)")
+	if _, err := r.RunCmd(exec.Command("sudo", "/bin/bash", "-c", "rm -rf /etc/cni/net.d/*")); err != nil {
```

Comment:
This removes the podman network, so that `podman run` doesn't work anymore.

We had similar issues with `kubeadm reset` before, that also destroyed podman.

Reply:
I changed the PR in a way that the podman CNI is not deleted. Instead a fake cni config is created IF the cni plugin is custom. This makes Pods using podNetwork waiting for the 3rd party CNI to come up.
Btw. Calico is suffering from the same problem. Following the operator manual (https://docs.projectcalico.org/getting-started/kubernetes/minikube) create coredns with a different IP than Calico should provide:
```
NAMESPACE         NAME                                       READY   STATUS    RESTARTS   AGE     IP              NODE       NOMINATED NODE   READINESS GATES
calico-system     calico-kube-controllers-6797d7cc89-pg2sg   1/1     Running   0          74s     10.244.0.3      minikube   <none>           <none>
calico-system     calico-node-q2qzd                          1/1     Running   0          74s     192.168.64.30   minikube   <none>           <none>
calico-system     calico-typha-54df7c5c64-qnqm9              1/1     Running   0          75s     192.168.64.30   minikube   <none>           <none>
kube-system       coredns-74ff55c5b-88zbx                    1/1     Running   0          2m47s   10.244.0.2      minikube   <none>           <none>
kube-system       etcd-minikube                              1/1     Running   0          2m57s   192.168.64.30   minikube   <none>           <none>
kube-system       kube-apiserver-minikube                    1/1     Running   0          2m57s   192.168.64.30   minikube   <none>           <none>
kube-system       kube-controller-manager-minikube           1/1     Running   0          2m57s   192.168.64.30   minikube   <none>           <none>
kube-system       kube-proxy-lppr4                           1/1     Running   0          2m48s   192.168.64.30   minikube   <none>           <none>
kube-system       kube-scheduler-minikube                    1/1     Running   0          2m57s   192.168.64.30   minikube   <none>           <none>
kube-system       storage-provisioner                        1/1     Running   0          3m2s    192.168.64.30   minikube   <none>           <none>
tigera-operator   tigera-operator-549bf46b5c-2jm8j           1/1     Running   0          114s    192.168.64.30   minikube   <none>           <none>
```

Reply:
my PR actually fixes Calico as well:
```bash
minikube start  --cni=/tmp/tigera-operator.yaml --extra-config=kubeadm.pod-network-cidr=192.168.0.0/16 --driver hyperkit  --container-runtime cri-o
kubectl create -f https://docs.projectcalico.org/manifests/custom-resources.yaml
```

```bash
NAMESPACE         NAME                                       READY   STATUS    RESTARTS   AGE     IP               NODE       NOMINATED NODE   READINESS GATES
calico-system     calico-kube-controllers-6797d7cc89-fshvj   1/1     Running   0          70s     192.168.120.65   minikube   <none>           <none>
calico-system     calico-node-dpvcq                          1/1     Running   0          70s     192.168.64.31    minikube   <none>           <none>
calico-system     calico-typha-66fd8b686-l8c6f               1/1     Running   0          70s     192.168.64.31    minikube   <none>           <none>
kube-system       coredns-74ff55c5b-65mq8                    1/1     Running   0          2m19s   192.168.120.66   minikube   <none>           <none>
kube-system       etcd-minikube                              1/1     Running   0          2m29s   192.168.64.31    minikube   <none>           <none>
kube-system       kube-apiserver-minikube                    1/1     Running   0          2m29s   192.168.64.31    minikube   <none>           <none>
kube-system       kube-controller-manager-minikube           1/1     Running   0          2m28s   192.168.64.31    minikube   <none>           <none>
kube-system       kube-proxy-pshfd                           1/1     Running   0          2m19s   192.168.64.31    minikube   <none>           <none>
kube-system       kube-scheduler-minikube                    1/1     Running   0          2m28s   192.168.64.31    minikube   <none>           <none>
kube-system       storage-provisioner                        1/1     Running   0          2m31s   192.168.64.31    minikube   <none>           <none>
tigera-operator   tigera-operator-549bf46b5c-p7mjr           1/1     Running   0          2m19s   192.168.64.31    minikube   <none>           <none>
```